### ISSUE TITLE
Issues setting up with docker-compose

### ISSUE THREAD

**Issue by samr28** [Yooooomi/your_spotify#10]
Hey I wanted to test this out so I modified the provided docker-compose file to try and use a different API endpoint (NGINX proxy). When I change it in the environment variables, it seems to still send requests to localhost:3001. The console output looks right so I'm not sure what went wrong here.

NGINX config:
```
server {

	server_name <removed>.com;

	location / {
		proxy_pass http://localhost:3002;
	}

	location /api {
		proxy_pass http://localhost:3001;
	}


    listen [::]:80;
    listen 80;

}
```

![image](https://user-images.githubusercontent.com/3688992/82857058-7cf6e780-9ed5-11ea-83dd-c2c3d99d2fb6.png)
![nginx](https://user-images.githubusercontent.com/3688992/82857172-da8b3400-9ed5-11ea-85c7-2ffe03067183.png)
![output](https://user-images.githubusercontent.com/3688992/82857249-19b98500-9ed6-11ea-9e71-c14f38c5198a.png)

**Comment by Yooooomi** [comment]
I don't see why it wouldn't work as is. Can you try going in incognito mode of your browser to check if it's not a cache issue ?

PS: It would be great if you could post your logging as plain text with your domain/credentials obfuscated

**Comment by samr28** [comment]
Looks like there may have been a caching issue (maybe on cloudflare's side because I tried in incognito yesterday). Now the requests are going to the right place but I'm getting a 502 from the API. Perhaps its a header issue? Do you have any header info in your nginx config?

![502](https://user-images.githubusercontent.com/3688992/82957105-e4fe0a00-9f77-11ea-9e82-a755bf764244.png)

```
$ docker-compose up
Starting your_spotify-mongo ... done
Starting your_spotify-web   ... done
Starting your_spotify-app   ... done
Attaching to your_spotify-web, your_spotify-mongo, your_spotify-app
your_spotify-mongo | 2020-05-26T22:34:56.596+0000 I  CONTROL  [main] Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'
your_spotify-mongo | 2020-05-26T22:34:56.608+0000 W  ASIO     [main] No TransportLayer configured during NetworkInterface startup
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] MongoDB starting : pid=1 port=27017 dbpath=/data/db 64-bit host=8155e04d7134
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] db version v4.2.6
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] git version: 20364840b8f1af16917e4c23c1b5f5efd8b352f8
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] allocator: tcmalloc
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] modules: none
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] build environment:
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten]     distmod: ubuntu1804
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten]     distarch: x86_64
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten]     target_arch: x86_64
your_spotify-mongo | 2020-05-26T22:34:56.612+0000 I  CONTROL  [initandlisten] options: { net: { bindIp: "*" } }
your_spotify-mongo | 2020-05-26T22:34:56.614+0000 I  STORAGE  [initandlisten] Detected data files in /data/db created by the 'wiredTiger' storage engine, so setting the active storage engine to 'wiredTiger'.
your_spotify-mongo | 2020-05-26T22:34:56.614+0000 I  STORAGE  [initandlisten] 
your_spotify-mongo | 2020-05-26T22:34:56.614+0000 I  STORAGE  [initandlisten] ** WARNING: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine
your_spotify-mongo | 2020-05-26T22:34:56.614+0000 I  STORAGE  [initandlisten] **          See http://dochub.mongodb.org/core/prodnotes-filesystem
your_spotify-mongo | 2020-05-26T22:34:56.614+0000 I  STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=7343M,cache_overflow=(file_max=0M),session_max=33000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000,close_scan_interval=10,close_handle_minimum=250),statistics_log=(wait=0),verbose=[recovery_progress,checkpoint_progress],
your_spotify-web | Setting API Endpoint to 'http://<URL>/api'
your_spotify-web | UPDATE AVAILABLE The latest version of `serve` is 11.3.1
your_spotify-web | INFO: Accepting connections at http://0.0.0.0:3000
your_spotify-mongo | 2020-05-26T22:34:57.197+0000 I  STORAGE  [initandlisten] WiredTiger message [1590532497:197033][1:0x7f3b50968b00], txn-recover: Recovering log 7 through 8
your_spotify-app | yarn run v1.22.4
your_spotify-mongo | 2020-05-26T22:34:57.237+0000 I  STORAGE  [initandlisten] WiredTiger message [1590532497:237231][1:0x7f3b50968b00], txn-recover: Recovering log 8 through 8
your_spotify-app | $ nodemon ./src/bin/www
your_spotify-mongo | 2020-05-26T22:34:57.298+0000 I  STORAGE  [initandlisten] WiredTiger message [1590532497:298446][1:0x7f3b50968b00], txn-recover: Main recovery loop: starting at 7/9984 to 8/256
your_spotify-mongo | 2020-05-26T22:34:57.376+0000 I  STORAGE  [initandlisten] WiredTiger message [1590532497:376948][1:0x7f3b50968b00], txn-recover: Recovering log 7 through 8
your_spotify-app | [nodemon] 2.0.3
your_spotify-app | [nodemon] to restart at any time, enter `rs`
your_spotify-app | [nodemon] watching path(s): *.*
your_spotify-app | [nodemon] watching extensions: js,mjs,json
your_spotify-app | [nodemon] starting `node ./src/bin/www`
your_spotify-mongo | 2020-05-26T22:34:57.425+0000 I  STORAGE  [initandlisten] WiredTiger message [1590532497:425025][1:0x7f3b50968b00], txn-recover: Recovering log 8 through 8
your_spotify-mongo | 2020-05-26T22:34:57.465+0000 I  STORAGE  [initandlisten] WiredTiger message [1590532497:465327][1:0x7f3b50968b00], txn-recover: Set global recovery timestamp: (0, 0)
your_spotify-mongo | 2020-05-26T22:34:57.474+0000 I  RECOVERY [initandlisten] WiredTiger recoveryTimestamp. Ts: Timestamp(0, 0)
your_spotify-mongo | 2020-05-26T22:34:57.478+0000 I  STORAGE  [initandlisten] Timestamp monitor starting
your_spotify-mongo | 2020-05-26T22:34:57.481+0000 I  CONTROL  [initandlisten] 
your_spotify-mongo | 2020-05-26T22:34:57.481+0000 I  CONTROL  [initandlisten] ** WARNING: Access control is not enabled for the database.
your_spotify-mongo | 2020-05-26T22:34:57.481+0000 I  CONTROL  [initandlisten] **          Read and write access to data and configuration is unrestricted.
your_spotify-mongo | 2020-05-26T22:34:57.481+0000 I  CONTROL  [initandlisten] 
your_spotify-mongo | 2020-05-26T22:34:57.485+0000 I  SHARDING [initandlisten] Marking collection local.system.replset as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.487+0000 I  STORAGE  [initandlisten] Flow Control is enabled on this deployment.
your_spotify-mongo | 2020-05-26T22:34:57.487+0000 I  SHARDING [initandlisten] Marking collection admin.system.roles as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.487+0000 I  SHARDING [initandlisten] Marking collection admin.system.version as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.488+0000 I  SHARDING [initandlisten] Marking collection local.startup_log as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.488+0000 I  FTDC     [initandlisten] Initializing full-time diagnostic data capture with directory '/data/db/diagnostic.data'
your_spotify-mongo | 2020-05-26T22:34:57.490+0000 I  SHARDING [LogicalSessionCacheRefresh] Marking collection config.system.sessions as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.490+0000 I  SHARDING [LogicalSessionCacheReap] Marking collection config.transactions as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.496+0000 I  NETWORK  [listener] Listening on /tmp/mongodb-27017.sock
your_spotify-mongo | 2020-05-26T22:34:57.496+0000 I  NETWORK  [listener] Listening on 0.0.0.0
your_spotify-mongo | 2020-05-26T22:34:57.496+0000 I  NETWORK  [listener] waiting for connections on port 27017
your_spotify-app | (node:38) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
your_spotify-mongo | 2020-05-26T22:34:57.925+0000 I  NETWORK  [listener] connection accepted from 172.20.0.4:47476 #1 (1 connection now open)
your_spotify-mongo | 2020-05-26T22:34:57.929+0000 I  NETWORK  [conn1] received client metadata from 172.20.0.4:47476 conn1: { driver: { name: "nodejs|Mongoose", version: "3.4.1|5.8.11" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "5.4.0-29-generic" }, platform: "Node.js v13.13.0, LE" }
your_spotify-app | (node:38) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead.
your_spotify-mongo | 2020-05-26T22:34:57.942+0000 I  SHARDING [conn1] Marking collection your_spotify.artists as collection version: <unsharded>
your_spotify-app | (node:38) DeprecationWarning: collection.count is deprecated, and will be removed in a future version. Use Collection.countDocuments or Collection.estimatedDocumentCount instead
your_spotify-mongo | 2020-05-26T22:34:57.950+0000 I  NETWORK  [listener] connection accepted from 172.20.0.4:47478 #2 (2 connections now open)
your_spotify-mongo | 2020-05-26T22:34:57.950+0000 I  NETWORK  [conn2] received client metadata from 172.20.0.4:47478 conn2: { driver: { name: "nodejs|Mongoose", version: "3.4.1|5.8.11" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "5.4.0-29-generic" }, platform: "Node.js v13.13.0, LE" }
your_spotify-mongo | 2020-05-26T22:34:57.951+0000 I  SHARDING [conn1] Marking collection your_spotify.albums as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.951+0000 I  SHARDING [conn2] Marking collection your_spotify.tracks as collection version: <unsharded>
your_spotify-mongo | 2020-05-26T22:34:57.952+0000 I  NETWORK  [listener] connection accepted from 172.20.0.4:47480 #3 (3 connections now open)
your_spotify-mongo | 2020-05-26T22:34:57.952+0000 I  NETWORK  [conn3] received client metadata from 172.20.0.4:47480 conn3: { driver: { name: "nodejs|Mongoose", version: "3.4.1|5.8.11" }, os: { type: "Linux", name: "linux", architecture: "x64", version: "5.4.0-29-generic" }, platform: "Node.js v13.13.0, LE" }
your_spotify-mongo | 2020-05-26T22:34:57.953+0000 I  SHARDING [conn3] Marking collection your_spotify.users as collection version: <unsharded>
your_spotify-app | info: Starting loop for 0 users
your_spotify-mongo | 2020-05-26T22:34:58.000+0000 I  SHARDING [ftdc] Marking collection local.oplog.rs as collection version: <unsharded>
your_spotify-app | info: Starting loop for 0 users
^CGracefully stopping... (press Ctrl+C again to force)
Stopping your_spotify-app   ... done
Stopping your_spotify-web   ... done
Stopping your_spotify-mongo ... done
```

**Comment by Yooooomi** [comment]
The 502 bad gateway issue is due to your nginx configuration imo. Did you try putting the slashes ?

**Comment by samr28** [comment]
My NGINX config is pasted below. It looks the same as yours so I'm not sure what could be wrong here.
```
server {

	server_name <URL>.com;

	location / {
		proxy_pass http://localhost:3002/;
	}

	location /api/ {
		proxy_pass http://localhost:3001/;
	}


    listen [::]:80;
    listen 80;

}

```

I get the error below from NGINX. It looks like it's routing the request to the right place (localhost:3001). Is there a test request that I could send from postman on the local machine to localhost:3001?
```
2020/05/26 18:46:34 [error] 187557#187557: *500 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: <IP>, server: <URL>.com, request: "POST /api/login HTTP/1.1", upstream: "http://127.0.0.1:3001/login", host: "<URL>.com", referrer: "http://<URL>.com/login"
```

I was thinking that maybe `API_ENDPOINT` should be `http://127.0.0.1:3001` for app because nginx should be routing /app/ to localhost:3001 so that means app should be listening on localhost:3001. This didn't seem to change anything though.

**Comment by Yooooomi** [comment]
If you `GET` / with Postman it should return `Hello !`.
As I remember, the `API_ENDPOINT` variable on the server is only here to redirect after Spotify Oauth.
You should try to troubleshoot the server by `GET`ing the / route from scratch, then adding your reverse proxy to see where it could go wrong.

**Comment by samr28** [comment]
Sorry for the delay, got busy with some other stuff. I actually just figured it out :). The problem was that the app wasn't listening to port 3001, it was listening to 8080. I thought it would remap based on that environment variable that gets passed in but now thinking about it, that doesn't make much sense lol.

So to fix it, I just changed the ports section for the app to: "3001:8080" so that 8080 gets mapped to 3001.
